Automattic/jetpack
# Implement the responses table for the new feedback dashboard

## Proposed changes:


This patch builds on #28790 by implementing the table for the responses list.

![Screenshot 2023-02-07 at 22 47 46](https://user-images.githubusercontent.com/8056203/217374518-0361aead-47ce-4591-b386-19062e1367c7.png)

I brought the `<Table />` component over from Crowdsignal for the most part to cover our needs here but it's pretty generic, and with some massaging and documentations, I think it could likely be bumped to `@automattic/jetpack-components` if that makes sense. But I'd say that's beyond the scope of this PR, even though I'm happy to work on that if y'all agree.

**Note:** This branch includes changes from the aforementioned PR as it's still awaiting review. The new commits start from `e40faf6`.

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






Enable the new feedback dashboard with these filters:

```
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

- Go to `/wp-admin/admin.php?page=jetpack-forms`.
- You should see the new dashboard layout featuring the new table populated with dummy data for the time being.
- Selecting individual items should work.
- "Select All" checkbox at the top should also work for selecting or deselecting all items at once.
- When only some items are selected, the "Select All" checkbox should display `-`.
